### ISSUE TITLE
Pitot cover should "break" pitot system, not ASI.

### ISSUE THREAD

**Issue by tigert** [Juanvvc/c172p-detailed#723]
Hey, nice new stuff again! :)

I noticed you added the pitot cover and correctly having an effect on the airspeed indicator when you forget to remove the cover. However I found two issues when testing this:

  * Forgetting the pitot cover on should not break the airspeed indicator, but the pitot system instead: `/systems/pitot/serviceable`
  * The worse issue seems to be that the airspeed indicator always works no matter what. If I read the code correctly, I think it pulls the airspeed value directly from the FDM's velocity variable, not via the pitot/static systems modelling:

   ```xml
 <animation>
  <type>rotate</type>
  <object-name>Needle</object-name>
  <property>fdm/jsbsim/velocities/vias-kts</property>
   ```

I am a bit confused in how our indicated-airspeed.xml plays into this, as I am not very familiar how all this comes together. But it might be that jsbsim and / or flightgear model pitot failures already?

**Comment by HHS81** [comment]
@tigert 

> `...<property>fdm/jsbsim/velocities/vias-kts</property>`

Interesting, didn't thought about this. Since I'm responsible for, I see now that taking the airspeed value from the FDM's velocity variable wasn't good. Indeed maybe better taking instruments/airspeed-indicator/....

Can you tell me, what will really happen when the pitot system is iced/ blocked?
Somewhere I read that it will show false values, but not simply nothing?

**Comment by onox** [comment]
Uhhh... @HHS81 you came with the idea to use `Systems/indicated-airspeed.xml`? :smile: 

> The worse issue seems to be that the airspeed indicator always works no matter what.

If `/sim/model/c172p/securing/pitot-cover-visible` is true, then `velocities/vias-kts` should be 0.0.

**Comment by onox** [comment]
@HHS81 You think we should use `/instrumentation/airspeed-indicator/indicated-speed-kt` instead of `velocities/vc-kts`? The former property does seem to react to `/systems/pitot/serviceable` (speed decreases while you climb if pitot failed and vice versa)

**Comment by HHS81** [comment]
>you came with the idea to use Systems/indicated-airspeed.xml?

> You mean to treat it as KCAS? We currently use velocities/vc-kts as the source.

Which is the source of this issue.

instead it should be instrumentation/airspeed-indicator/true-speed-kt as the source for uncorrected indicated-airspeed. When piot fails, true-speed-kt will fail and with that indicated-aispeed as well.

**Comment by onox** [comment]
`Systems/indicated-airspeed.xml` is determine KIAS based on KCAS, so why use KTAS to determine KIAS?

**Comment by onox** [comment]
The idea of using `Systems/indicated-airspeed.xml` was because `indicated-speed-kt` should be KIAS, but acted as KIAS. So we added our own table to convert to KIAS, but in the process we were using the KCAS from JSBSim, instead of `indicated-speed-kt`.

So we should use `indicated-speed-kt`, not `true-speed-kt`?

**Comment by HHS81** [comment]
@onox: No, no, no!!

the ASI on cessnas will never show the real indicated-airspeeds. They will always show errors. This errors are due the position of the pitot.
But FlightGear assumes that the pitot is a perfect pitot and sows real values. Which is not realistic.

So the table changes the correct-IAS from FlightGear back to uncorrect IAS from reallife. Just the source I used was not correct, because it does not depend on pitot-system and can't fail.

So source should be: `instrumentation/airspeed-indicator/true-speed-kt`

Clear?

**Comment by onox** [comment]
correct-IAS = KCAS and uncorrect IAS = KIAS? Or am I misunderstanding KIAS and KCAS?

**Comment by onox** [comment]
Also the POH uses KCAS in their tables, so I still don't understand why you want to use KTAS :worried:

**Comment by HHS81** [comment]
*sigh*

You making me mad. Yes, KTAS would be wrong. Sorry.

The table shows KCAS- calibrated or also called corrected airspeed. 
We are using JSBSim v**c**-kts. C stands for calibrated/ corrected. But this jsbsim values doesn't depend on the pitot. That's the problem

**Comment by onox** [comment]
> But this jsbsim values doesn't depend on the pitot.

I know that.

So you agree we should be using a KCAS from FG that reacts to failure of the pitot system?

**Comment by HHS81** [comment]
> So you agree we should be using a KCAS from FG that reacts to failure of the pitot system?

If there is a KCAS that depends on pitot/static systems , yes.

**Comment by onox** [comment]
There is: `/instrumentation/airspeed-indicator/indicated-speed-kt`

1. It reacts to `/systems/pitot/serviceable`

2. It assumes a perfect pitot, no error. Thus it is practically a KCAS

**Comment by HHS81** [comment]
> It assumes a perfect pitot, no error. Thus it is practically a KCAS

I would say yes.

**Comment by onox** [comment]
Okay, then I'll create a branch with a fix. :smiley:

**Comment by HHS81** [comment]
@onox 
you have to insert  `<property>/instrumentation/airspeed-indicator/indicated-speed-kt</property>`
between `<system name="indicated-airspeed">` 
and  `<channel name="indicated airspeed">`in the file, otherwise you will get an error message.
`

**Comment by onox** [comment]
:+1:

**Comment by HHS81** [comment]
hmmm.... there is a mismatch between `vc-kts` and `/instrumentation/airspeed-indicator/indicated-speed-kt`.
So actually `/instrumentation/airspeed-indicator/indicated-speed-kt` seems to show some error. But it is small.... hmmmm

**Comment by onox** [comment]
That might be fixable in `Systems/systems.xml`.

**Comment by onox** [comment]
I've pushed `bug-723`. We still could try to make sure that the very small error in `indicated-speed-kt` becomes 0.

**Comment by onox** [comment]
The airspeed indicator seem to apply a low pass filter:

```cpp
 91 void
 92 AirspeedIndicator::update (double dt)
 93 {
 94     if (!_serviceable_node->getBoolValue()) {
 95         return;
 96     }
 97 
 98     double pt = _total_pressure_node->getDoubleValue() ;
 99     double p = _static_pressure_node->getDoubleValue() ;
100     double qc = ( pt - p ) * SG_INHG_TO_PA ;  // Impact pressure in Pa, _not_ to be confused with dynamic pressure!!!
101 
102     // Now, reverse the equation (normalize impact pressure to
103     // avoid "nan" results from sqrt)
104     qc = std::max( qc , 0.0 );
105     // Calibrated airspeed (using compressible aerodynamics) based on impact pressure qc in m/s
106     // Using calibrated airspeed as indicated airspeed, neglecting any airspeed indicator errors.
107     double v_cal = sqrt( 7 * SG_p0_Pa/SG_rho0_kg_p_m3 * ( pow( 1 + qc/SG_p0_Pa  , 1/3.5 )  -1 ) );
108 
109                             // Publish the indicated airspeed
110     double last_speed_kt = _speed_node->getDoubleValue();
111     double current_speed_kt = v_cal * SG_MPS_TO_KT;
112     double filtered_speed = fgGetLowPass(last_speed_kt,
113                                              current_speed_kt,
114                                              dt * RESPONSIVENESS);
115     _speed_node->setDoubleValue(filtered_speed);
```

**Comment by onox** [comment]
I'll recompile FG with `filtered_speed = current_speed_kt` and see if that would fix it.

**Comment by tigert** [comment]
The airspeed instrument shows IAS - indicated airspeed (well, that's why it is called "indicated" airspeed, as that's what is shown by the "airspeed indicator" gauge).

TAS is true airspeed that is calculated (corrected) from the IAS by taking your altitude and temperature (basically pressure altitude) into account. The airspeed gauge has no idea about these. Some instruments have a TAS correcting ring that you can set manually by setting your density altitude and it has a second dial where you can then read your true airspeed (TAS). This is significant because the higher you go, the faster you cruise, even though your airspeed indicator shows the same indicated speed.

The xml lookup table that @HHS81 meant likely is a "error mapper" that creates the pitot tube inaccuracies caused by flaps (pitot is at a different angle at different flap settings, so the airspeed error is not constant)

This explains the system and various errors and terms about the pitot-static-system: http://www.allstar.fiu.edu/aero/PSSI.htm - interesting stuff for reading. 

The above and this explains what happens when the pitot tube is blocked:
https://en.wikipedia.org/wiki/Pitot-static_system#Blocked_pitot_tube 

_A blocked pitot tube will cause the airspeed indicator to register an increase in airspeed when the aircraft climbs, even though actual airspeed is constant. This is caused by the pressure in the pitot system remaining constant when the atmospheric pressure (and static pressure) are decreasing. In reverse, the airspeed indicator will show a decrease in airspeed when the aircraft descends._

Basically the way an airspeed indicator works is that it takes the static air pressure (a hole on the side of the fuselage usually) and the pitot tube pressure (air pushed to the tube due to the aircraft moving forward) and compares the difference, like two people blowing into both ends of glass tube trying to push a ball in the middle. Block either end and you see what happens.

We could check how the SenecaII does this, and how well this is modelled in FG.

**Comment by onox** [comment]
> A blocked pitot tube will cause the airspeed indicator to register an increase in airspeed when the aircraft climbs, even though actual airspeed is constant. This is caused by the pressure in the pitot system remaining constant when the atmospheric pressure (and static pressure) are decreasing. In reverse, the airspeed indicator will show a decrease in airspeed when the aircraft descends.

Checkout `bug-723` and you will see that it exactly works like that.
